### ISSUE TITLE
no-restricted-globals incorrectly reports shadowed function parameter named event

### ISSUE THREAD

**Issue by doberkofler** [oxc-project/oxc#23862]
### Oxlint Version

`1.71.0`

This appears to be a regression from `1.70.0`.

### Environment

- Node: `v26.2.0`
- npm: `11.17.0`
- OS: macOS

### Minimal Reproduction

`oxlint.config.ts`:

```ts
import {defineConfig} from 'oxlint';

export default defineConfig({
	rules: {
		'no-restricted-globals': [
			'error',
			{name: 'event', message: 'The event object is globally available but causes problems when used.'},
		],
	},
});
```

`index.ts`:

```ts
document.addEventListener('click', (event) => {
	const target = event.target;
	console.log(target);
});
```

Run:

```sh
npx oxlint index.ts
```

### Actual Behavior

Oxlint reports `event` as a restricted global:

```text
eslint(no-restricted-globals): Unexpected use of 'event'. The event object is globally available but causes problems when used.

document.addEventListener('click', (event) => {
	const target = event.target;
	               ^^^^^
});
```

### Expected Behavior

No error should be reported.

The `event` identifier is a local function parameter and shadows the browser global `event`. This is also what the diagnostic help text suggests:

```text
Use a local variable or function parameter instead of the restricted global.
```

### Notes

This currently causes many false positives in event handlers such as DOM, jQuery, React, and hotkeys callbacks.

The rule should only report unresolved/global references to `event`, for example:

```ts
const target = event.target;
```

It should not report locally bound identifiers:

```ts
document.addEventListener('click', (event) => {
	const target = event.target;
});
```

**Comment by Boshen** [comment]
Fixed by #23736 (in `main`, landed just after the v1.71.0 tag you were on). Parameters/bindings that shadow a restricted global are no longer flagged.

Verified on current `main`: with `"no-restricted-globals": ["error", "event"]`, both `function handler(event) { return event.target; }` and `addEventListener('click', (event) => event.target)` produce **0** `no-restricted-globals` errors, while a bare `event.target` is still correctly flagged. Closing as resolved — please reopen if you still see this on the latest release.
